### PR TITLE
ref(widget-builder): Add common component for section header

### PR DESCRIPTION
**Pull request by narsaynorath**
Just avoids some duplicate code and makes the section header with the tooltip reusable.

### CODE REVIEW

File: static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx [hunk 1]
```diff
@@ -79,8 +74,3 @@ const FeatureBadgeAlignmentWrapper = styled('div')`
     top: -1px;
   }
 `;
-
-const Header = styled('h6')`
-  font-size: ${p => p.theme.fontSizeLarge};
-  margin-bottom: ${space(2)};
```
Comment (nikkikapadia) [inline static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx:85]: are we able to keep this different bottom margin for the dataset header? it looks a little better and closer to the design 👀

**Review by nikkikapadia — APPROVED** [review]
Nice been meaning to do this one. Just have one smol request before you merge.

File: static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx (hunk 1 above)
Comment (narsaynorath) [reply static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx:85]: Ohh I didn't realize it was a bit more! hmmm let's see

Comment (narsaynorath) [reply static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx:85]: I added a way to apply styles using the styled component for this. It's a bit hacky but for now I don't think it needs to change much

Comment (nikkikapadia) [reply static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx:85]: amazing thanks!

**Pull request merged**